### DISCUSSION TITLE
Flock naming

### DISCUSSION THREAD

**Discussion by FlameArrow57** [stonepillars/goon-flock#215] [Ideas]
Discussion for changing flock names.

Flockmind is currently named as cv.cv (ex. pa.li).
Flocktraces are currently named as Cv.v (ex. Pa.o).
Flockdrones are currently named as cv.cv.cv (ex. pe.fu.re).
Flockbits are currently named as C.rand(10, 99).rand(10, 99) (ex. Q.45.97)

The names are for the most part fine I think, it's an interesting naming scheme. Only thing is that it doesn't really follow any hierarchy pattern that one might expect. It's not clear looking at the names of things from different flocks and knowing they are from different flocks.

Suggested new naming scheme:
Flockmind naming is the same, cv.cv (ex. pa.li).
Flocktraces are now [flockmind].cv (ex. pa.li.re), suggested by stonepillar.
Flockdrones are now [flockmind].rand(1, 1000) (ex. pa.li.327).
Flockbits are now [flockdrone].rand(1, 99) (ex. pa.li.327.8).

**Comment by amylizzle** [comment]
A flockdrone can only split into 3 bits, so rand(1,99) is probably overkill.
also I'm not so keen on the numbers, I kinda like the simple syllables pattern - maybe drones could be pa.li.[aaa-zzz] and bits could be [drone].[a-z]?

**Reply by stonepillars** [reply]
Could append the number to the _end_ of the flockdrone's syllabled name. i.e. pa.li.re2 (meaning bit 2 from drone 're' from flock 'pa.li')
Though using syllables for drones does not distinguish between drones and flocktraces.

**Reply by TobleroneSwordfish** [reply]
Maybe write the numbers in hex, so pa.li.2F or ro.pi.A2 would be drones

**Reply by amylizzle** [reply]
lowercase tho, I know it's not proper but it's more _aesthetic_

**Comment by TobleroneSwordfish** [comment]
I like the traces being linked to the parent flockmind, because it means you can easily refer to each other with a single syllable, like `ro, what are you doing?` instead of trying to remember multiple arbitrary syllables.

**Reply by stonepillars** [reply]
I suddenly have the idea for being able to name your drones.
OH NO BURGERBIRD DIED

**Comment by pali6** [comment]
Making the names hierarchical makes sense but would imo be more confusing and less useful in-game, especially considering that presumably there usually won't be more than one flock. When everything in your chat starts with pa.li it'll just start to blend together I'd say.

**Reply by amylizzle** [reply]
That is a very good point. It's already pretty rough figuring out what's going on when your chat is spammed with "the weird doodad absorbs the wrench" and "drone awakened, assessing situation"

**Reply by amylizzle** [reply]
Maybe do it as prefix instead of suffix - a.fu.pa.li for a flockbit for example

**Reply by stonepillars** [reply]
We can separate the relative name from the fully qualified flock name. So drone `a2` could be what appears in chat, but  it's actual name is `pa.li.a2`.

**Comment by amylizzle** [comment]
Also we should check for uncool words. Not to put too fine a point on it, but if the first syllable is "ni" or "fa" we should probably prevent the next syllable from starting with g.

**Comment by FlameArrow57** [comment]
Ok so based on what was talked about, summarizing:

Going to remove the hierarchy naming.

Not sure if hex would be good. First, though hex is computational based, not sure if it's in theme of Flock to use base 16?

Suggesting some narrowed options:

Flockmind name kept: cv.cv (ex. re.tu)

 1. 
Flocktrace name: cv.v (ex. mo.p)
Flockdrone name kept: cv.cv.cv (ex. ma.gi.wa) (alternative, one of the last two letters is a number, ex. ma.gi.2r)
Flockbit name: [drone-name].[1-3] (ex. ma.gi.wa.3) (or hypen, ex. ma.gi.wa-3)

2.
Flocktrace name: cv.cv.cv (ex. ma.gi.wa)
Flockdrone name: cv.cv.rand(1, 999) (ex. re.li.789)
Flockbit name: [drone-name].[1-3] (ex. re.li.789.2) (or hypen, ex. re.li.789-2)

Alternative bit names for both cases: Considering flockbits are meant to be mindless drones, I feel like it could be their naming doesn't have to include numbers 1-3, and could be random numbers, ex. 1. fe.li.ru.38 or fe.li.ru-38 and 2. re.li.789.39 or re.li.789-39. Hex could be more in theme considering the name "Flockbit," but that starts to feel like maybe the Flockbit name is blending in with the Flockdrone name, ex. 1. fe.li.ru.2f or fe.li.ru-2f and 2. re.li.789.f4 or re.li.789-f4.

3.
Names kept except for flocktrace:
Flocktrace name is cv.v instead of Cv.v (ex. te.q vs Te.q)
